gateway/assets/script/bootstrap.js

```diff
@@ -4,6 +4,16 @@
 
 var app = angular.module('faasGateway', ['ngMaterial', 'faasGateway.funcStore']);
 
+app.directive("readyInput", function() {
```

Comment:
We should avoid using `directive`s if we aren't dynamically manipulating the DOM (e.g. adding/removing with `jqlite`) . We should use `component` instead and isolate the `scope` to reduce shared state.

In this case, it might be good enough to just add the `md-input-container` in the existing `html` without creating another directive (component), imo.

Reply:
I think it could be inlined again now. I had issues with ng-model now allowing a method to be called.

Reply:
hmm... shared `$scope` is an anti-pattern in AngularJS. I can add an example tonight with a `component`.  

> I had issues with ng-model now allowing a method to be called

I would like to hear what happened here. Would like to reproduce the problem.

I'm going out today so it'll be at least 12 hours from now 😉